Library/Homebrew/formulary.rb

```diff
@@ -917,30 +917,41 @@ class FromAPILoader < FormulaLoader
     def self.try_new(ref, from: T.unsafe(nil), warn: false)
       return if Homebrew::EnvConfig.no_install_from_api?
       return unless ref.is_a?(String)
-      return unless (name = ref[HOMEBREW_DEFAULT_TAP_FORMULA_REGEX, :name])
-      if !Homebrew::API::Formula.all_formulae.key?(name) &&
-         !Homebrew::API::Formula.all_aliases.key?(name) &&
-         !Homebrew::API::Formula.all_renames.key?(name)
-        return
+
+      ref = if (name = parse_name(ref))
```

Comment:
Thoughts on naming this something like:

```suggestion
      ref = if (name = parse_homebrew_core_name(ref))
```

Or something like that? That way it's clear that the "parsing" that's happening is only within the homebrew/core tap?

Reply:
Maybe `parse_core_name` here and `parse_core_token` in `Cask::CaskLoader`?